### PR TITLE
DynamicProtobufSerializer: Debug message for serializer error

### PR DESCRIPTION
**Pull request by hisundar**
## Overview

If a table was opened with a different type but carries entries of different types, then we should be able to catch and log this error.
## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java [hunk 1]
```diff
@@ -151,6 +151,9 @@ private FileDescriptor getDescriptor(String name) throws DescriptorValidationExc
             return fileDescriptorMap.get(name);
         }
 
+        if (!fdProtoMap.containsKey(name)) {
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java]: Is it possible to collapse this with fileDescriptorMap?

It's also seems a little bit strange to call getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, REGISTRY_TABLE_NAME) inside the constructor I feel like its cleaner to open that map and extract the serializers from a higher layer.

File: runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java [hunk 2]
```diff
@@ -151,6 +151,9 @@ private FileDescriptor getDescriptor(String name) throws DescriptorValidationExc
...
+            throw new SerializerException("DynamicProtobuf type "+name+" was never seen in registry");
```
Comment (xnull) [inline runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java]: code reformat

File: runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java (hunk 1 above)
Comment (hisundar) [reply runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java]: I have been exploring this, looks like right there isn't an obvious way to directly convert from a type descriptor to a type. That is why we are having this extra layer of indirection to capture the File itself. It definitely needs more digging that a debugging patch.

File: runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java (hunk 2 above)
Comment (hisundar) [reply runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java]: done. thanks

File: runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java [hunk 3]
```diff
@@ -151,6 +151,11 @@ private FileDescriptor getDescriptor(String name) throws DescriptorValidationExc
             return fileDescriptorMap.get(name);
         }
 
+        if (!fdProtoMap.containsKey(name)) {
+            log.error(this.toString());
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java]: Can you add a log error message here, maybe log.error("cannot deserialize object {}", this, exception)

File: runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java [hunk 4]
```diff
@@ -284,5 +291,26 @@ public void serialize(Object o, ByteBuf b) {
             throw new SerializerException(ie);
         }
     }
+
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
```
Comment (xnull) [inline runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java]: it will be very hard to parse these kinds of messages in log aggregation platforms like ELK.
Can you use one string for that?

Comment (hisundar) [reply runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java]: hi @xnull sorry I missed this comment before merge. I would like to address it. Could you please share details on what is the ELK format for strings?

Comment (xnull) [reply runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java]: It's just a best practice, never use more than one line in the log. In the other case, you lose the opportunity to parse your logs as a whole thing
